src/mapping.ts

```diff
@@ -10,18 +10,36 @@ import { tokens, trades, orders, users } from "./modules"
 export function handleInteraction(event: Interaction): void {}
 
 export function handleOrderInvalidated(event: OrderInvalidated): void {
+
   let orderId = event.params.orderUid.toHexString()
-  orders.invalidateOrder(orderId)
+  let timestamp = event.block.timestamp
+
+  let order = orders.invalidateOrder(orderId, timestamp)
+
+  order.save()
 }
 
-export function handlePreSignature(event: PreSignature): void {}
+export function handlePreSignature(event: PreSignature): void {
+
+  let orderUid = event.params.orderUid.toHexString()
+  let ownerAddress = event.params.owner
+  let owner = ownerAddress.toHexString()
+  let timestamp = event.block.timestamp
+
+  let order = orders.preSig(orderUid, owner, timestamp)
```

Comment:
you have to be very thorough checking the parameters of the contract events. Here you are missing one important parameter. There's a boolean that can be to sign or to remove a signature.

We want to keep all the information and be as close as possible as the protocol/contracts.

Reply:
EDIT: amending my comment, I think you added this param initially and removed addressing some comment. I think we should either use it to set the timestamp or include it in the entity as i suggest in one comment.

I will mark this conversation as resolved, just because there's similar conversations in other threads.